From c6ecc108716b12b746d1a6e3f19602a47be87b76 Mon Sep 17 00:00:00 2001
From: Patrick Ohly <patrick.ohly@intel.com>
Date: Wed, 25 Mar 2015 10:41:13 -0700
Subject: [PATCH] systemd: stop using compat libs

libsystemd-journal are considered obsolete in systemd since 2.09 and may not
be available (not compiled by default).

The code works fine with the current libsystemd, so just use that. While at
it, use pkg-config to find the right flags.

Signed-off-by: Patrick Ohly <patrick.ohly@intel.com>
---
 Makefile.am  | 4 ++--
 configure.ac | 2 ++
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 51da5ee..00409dc 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -10,7 +10,7 @@ libdlog_la_SOURCES =  \
 	log.c \
 	log-common.c
 
-libdlog_la_LIBADD = $(TZPLATFORM_CONFIG_LIBS) -lsystemd-journal
+libdlog_la_LIBADD = $(TZPLATFORM_CONFIG_LIBS) $(SYSTEMD_LIBS)
 
 bin_PROGRAMS= dlogctrl
 
@@ -20,7 +20,7 @@ dlogctrl_SOURCES = \
 
 dlogctrl_LDADD = $(TZPLATFORM_CONFIG_LIBS)
 
-dlogctrl_CFLAGS = $(AM_CFLAGS)
+dlogctrl_CFLAGS = $(AM_CFLAGS) $(SYSTEMD_CFLAGS)
 
 # conf file
 pkgconfigdir = $(libdir)/pkgconfig
diff --git a/configure.ac b/configure.ac
index 9330ec2..4621a2b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -13,6 +13,8 @@ AC_CONFIG_HEADERS([config.h])
 
 PKG_CHECK_MODULES([TZPLATFORM_CONFIG], [libtzplatform-config])
 
+PKG_CHECK_MODULES([SYSTEMD], [libsystemd])
+
 # Checks for programs.
 m4_ifdef([AM_PROG_AR], [AM_PROG_AR])
 AC_PROG_CC
-- 
1.8.4.5

